drivers/bridge/setup_ip_tables.go

```diff
@@ -13,14 +14,61 @@ const (
 	DockerChain = "DOCKER"
 )
 
-func (n *bridgeNetwork) setupIPTables(config *networkConfiguration, i *bridgeInterface) error {
+func setupIPChains(config *configuration) (*iptables.ChainInfo, *iptables.ChainInfo, error) {
 	// Sanity check.
 	if config.EnableIPTables == false {
-		return IPTableCfgError(config.BridgeName)
+		return nil, nil, fmt.Errorf("Cannot create new chains, EnableIPTable is disabled")
 	}
 
 	hairpinMode := !config.EnableUserlandProxy
 
+	natChain, err := iptables.NewChain(DockerChain, iptables.Nat)
+	if err != nil {
+		return nil, nil, fmt.Errorf("Failed to create NAT chain: %s", err.Error())
+	}
+	defer func() {
+		if err != nil {
+			if err := iptables.RemoveExistingChain(DockerChain, iptables.Nat); err != nil {
+				logrus.Warnf("Failed on removing iptables NAT chain on cleanup: %v", err)
+			}
+		}
+	}()
+
+	// TODO(banix): Move the following block; keep this function for just creating the chains
+	natChain, err = iptables.ProgramChain(natChain, hairpinMode)
+	if err != nil {
+		return nil, nil, fmt.Errorf("Failed to program NAT chain: %s", err.Error())
+	}
+
+	filterChain, err := iptables.NewChain(DockerChain, iptables.Filter)
+	if err != nil {
+		return nil, nil, fmt.Errorf("Failed to create FILTER chain: %s", err.Error())
+	}
+	defer func() {
```

Comment:
I don't think this defer is needed, given nothing else gets programmed/created after the filterChain.
If filter chain fails to be created, we just return and execute the previous defer for the natChain removal.